include/core/boundary_conditions.h

```diff
@@ -745,6 +745,129 @@ namespace BoundaryConditions
     prm.leave_subsection();
   }
 
+  /**
+* @brief This class manages the boundary conditions for the Cahn-Hilliard solver
+* It introduces the boundary functions and declares the boundary conditions
+* coherently.
+*  - if bc type is "dirichlet" (Dirichlet condition), "value" is the
+* double passed to the deal.ii ConstantFunction
+
+  */
+
+  template <int dim>
+  class CahnHilliardBoundaryConditions : public BoundaryConditions<dim>
+  {
+  public:
+    void
+    declareDefaultEntry(ParameterHandler &prm, unsigned int i_bc);
+    void
+    declare_parameters(ParameterHandler &prm);
+    void
+    parse_boundary(ParameterHandler &prm, unsigned int i_bc);
+    void
+    parse_parameters(ParameterHandler &prm);
+  };
+
+  /**
+   * @brief Declares the default parameters for a boundary condition id i_bc
+   * i.e. Dirichlet condition (ConstantFunction) with value 0
+   *
+   * @param prm A parameter handler which is currently used to parse the simulation information
+   *
+   * @param i_bc The boundary condition id.
+   */
+  template <int dim>
+  void
+  CahnHilliardBoundaryConditions<dim>::declareDefaultEntry(
+    ParameterHandler &prm,
+    unsigned int      i_bc)
+  {
+    return;
```

Comment:
I don't understand why the return is first. Is it temporary?

Reply:
The order should be reversed indeed